domain/schema/model/sql/0022-resource.sql

```diff
@@ -0,0 +1,146 @@
+CREATE TABLE resource_origin_kind (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_resource_origin_name
+ON resource_origin_kind (name);
+
+INSERT INTO resource_origin_kind VALUES
+(0, 'uploaded'),
+(1, 'store');
+
+CREATE TABLE resource_state (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_resource_state
+ON resource_state (name);
+
+-- Resource state values:
+-- Available is the application resource which will be used by any units
+-- at this point in time.
+-- Potential indicates there is a different revision of the resource available
+-- in a repository. Used to let users know a resource can be upgraded.
+INSERT INTO resource_state VALUES
+(0, 'available'),
+(1, 'potential');
+
+CREATE TABLE resource_hash_kind (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_resource_hash_kind
+ON resource_hash_kind (name);
+
+INSERT INTO resource_hash_kind VALUES
+(0, 'fingerprint');
```

Comment:
fingerprint isn't a valid hash kind, what I mean was to write the exact hash in use i.e. `sha512-384`

https://github.com/juju/utils/blob/5f18edb2a126f0e32dc646d8ce80489cdc40685a/hash/hash.go#L60

Reply:
Also, we should bring in the utils/hash package into the internal package. One less `util` dependency.

Reply:
made the schema change.